src/main/joyride/scripts_handler.cljs

```diff
@@ -0,0 +1,190 @@
+(ns joyride.scripts-handler
+  (:require ["path" :as path]
+            ["vscode" :as vscode]
+            [joyride.config :as conf]
+            [joyride.constants :as const]
+            [joyride.db :as db]
+            [joyride.sci :as jsci]
+            [joyride.utils :as utils :refer [cljify jsify]]
+            [promesa.core :as p]
+            [sci.core :as sci]))
+
+(defn find-script-uris+
+  "Returns a Promise that resolves to JS array of `vscode.Uri`s
+   for the scripts files found in `base-path`/`script-folder-path`
+   
+   Will use `vscode/workspace` API for it if there is a workspace
+   root, otherwise it uses direct filesystem access. (Probably means
+   it is only Remote friendly in the case with a workspace root.)"
+  [base-path script-folder-path]
+  (if vscode/workspace.rootPath
+    (p/let [glob (vscode/RelativePattern. base-path (path/join script-folder-path "**" "*.cljs"))
+            script-uris (p/->> (vscode/workspace.findFiles glob)
+                               cljify
+                               (sort-by #(.-fsPath ^js %)))]
+      (jsify script-uris))
+    (p/let [crawl-path (path/join base-path script-folder-path)
+            glob "**/*.cljs"
+            script-uris (utils/find-fs-files+ crawl-path glob)]
+      (jsify script-uris))))
+
+(defn strip-abs-scripts-path [abs-scripts-path abs-path]
```

Comment:
What does this function do? Perhaps add a docstring, because it isn't at all obvious to me.